modules/broker-core/src/main/java/org/wso2/broker/core/util/TraceField.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.broker.core.util;
+
+/**
+ * Custom tracing field object
+ */
+public class TraceField {
+
+    private static final String DELIMITER = ": ";
+
+    private final String field;
+
+    public TraceField(String key, Object value) {
+        this.field = key + DELIMITER + value;
```

Comment:
Maybe we should use a string builder instead of concatenating Strings.

Reply:
Used separate key values in TraceField data structure. Passed to the MessageTracer and then used String Builder to concatenate the strings.